crates/vault-registry/src/types.rs

```diff
@@ -444,7 +447,20 @@ impl<T: Config> RichVault<T> {
     }
 
     pub(crate) fn slash_to_liquidation_vault(&mut self, amount: Collateral<T>) -> DispatchResult {
-        ext::staking::slash_stake::<T>(&self.id(), amount)?;
+        let vault_id = self.id();
+
+        // get the collateral supplied by the vault (i.e. excluding nomination)
+        let collateral = Pallet::<T>::compute_collateral(&vault_id)?;
+        let (to_withdraw, to_slash) = amount
```

Comment:
Shouldn't we use a saturating sub here? I.e. if the vault has _some_ collateral, but less than `amount`, should we not slash whatever it has?

Reply:
I think this is already handled, if the vault has less collateral than the amount then the remainder is taken from the nominators. Implicitly the `amount` can not be greater than the collateral supplied by a vault and it's nominators.

Reply:
Ah woops you are right